### PR TITLE
feat: Improved Notebooks UX

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Continued work for Notebooks

## Changes

- [x] Fixes the animation so that the parent immediately snaps but the notebook animates
- [x] Delay the rendering of custom components to give a more slick feeling
- [x] Better full screen view
- [x] Lazy load off screen components
- [x] Enforce a fixed h1 title at the top
- [x] Better listing and creation flow to mimic real API
- [x] Delayed saving via localstorage
- [x] UI for finding and loading individual notebooks
- [x] Derive title from header
- [x] Fix issues to do with no longer using the page full screen scrolling. Fixing the body height might not have been the best approach

## Follow up TODOs
https://github.com/PostHog/posthog/issues/15680

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/__snapshots__/filters-person-property-select--sortable.png
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/__snapshots__/filters-person-property-select--sortable.png:1]: Darn!!1ß02032∂@

File: playwright/e2e-vrt/README.md [hunk 1]
```diff
@@ -60,3 +60,7 @@ Troubleshoot by viewing the Playwright report: Click on "Details" next to the fa
 ### Your locally generated images, don't verify CI checks
 
 GitHub is running tests against a temporary merge commit (to ensure tests still pass after being merged), meaning any changes currently in master will be present in the images generated in the CI run. If you suspect this is the case, merge master into your branch and push again.
+
+### The screenshots look wrong and I want to debug it live
+
+If you run any playwright command with `PWDEBUG=1` then it will run in "headfull" mode so that you can see the browser and look into what is going on
```
Comment (pauldambra) [inline playwright/e2e-vrt/README.md:66]: ✨

**Review by pauldambra — APPROVED** [review]
LGTM

A few things from clicking around but no need for them to be blocking

# add to notebook not working on feature flag page



# banner hides notebook controls



# add to notebook replacing content

**Comment by benjackwhite** [comment]
@pauldambra those look like the old UI 🤔 did you actually check out the branch :D

**Comment by pauldambra** [comment]
> @pauldambra those look like the old UI 🤔 did you actually check out the branch :D

🙈 yep, but I didn't check the JS built

homer-hedge

**Pull request merged**